### DISCUSSION TITLE
Humanoid model: Limb positions initialization

### DISCUSSION THREAD

**Discussion by ThoChat** [PedestrianDynamics/jupedsim#1428] [Q&A] [answered]
Hi there,
I'm setting up a model that includes agents limb positions. 
For now, the only limb is the head.
The model works, the head moves according to the model. 
However, the head_position on the **first frame** is not associated with the position given during the initialization of the agent but remains at the default value: (0,0). 
When populating my simulation I use this :
  simulation.add_agent(
        jps.HumanoidModelV0AgentParameters(
            journey_id=journey_id,
            stage_id=exit_id,
            position=pos,
            head_position=pos,
            desiredSpeed=v0,
        )
    )

It's probably coming from the initialization, but to me, this seems to be set properly since the position is given in the creation of the instances. Does anyone have an idea?

**Comment by Ozaq** [comment]
Can you post a link to your branch, from what you said it sounds like you did not pass on the initial value. Easy error to make with our many layers.

I can have a quick look at it during lunch

**Reply by chraibi** [reply]
I think this one: https://github.com/ThoChat/jupedsim/tree/Humanoid-Pedestrian-Paradigm

**Reply by ThoChat** [reply]
Yes indeed, but I can't figure out where though.
Here's the link:
https://github.com/ThoChat/jupedsim/tree/Humanoid-Pedestrian-Paradigm

Thanks a lot for your help!

**Reply by Ozaq** [reply]
I can't see from the code alone where the value is lost, do you have a link to the example what you are trying to run? I can look at it tonight.

**Reply by ThoChat** [reply]
Here is the code I use to test the model.
https://github.com/ThoChat/jupedsim_test_sandbox
The file test.py creates a simulation based on the scenario of the "corner" notebook.
The file sqlite_loader_moded_pepy_fun.py provides a modified function from pedpy that includes the head trajectory in the "data" argument of the TrajectoryData object created by the sqlite loader.

**Reply by chraibi** [reply]
Hi @Ozaq,
I tried with this [code](https://fz-juelich.sciebo.de/s/HDeShbdYWBKdjQw)

I get the following:
```
agent.model.desiredSpeed = 1.0 should be 1.0  agent.model.height = 0.0 should be 1.0
```

we could not identify where the leak. Could you please take a look?

**Reply by Ozaq** [reply, marked as answer]
I finally had some time to check your code. You forgot to pass on the parameters to the add agent call. Apply this patch to your code to have all parameters available

```text
diff --git a/libjupedsim/src/simulation.cpp b/libjupedsim/src/simulation.cpp
index 8c1814096..8048bbb56 100644
--- a/libjupedsim/src/simulation.cpp
+++ b/libjupedsim/src/simulation.cpp
@@ -348,7 +348,21 @@ JPS_AgentId JPS_Simulation_AddHumanoidModelV0Agent(
                 parameters.agentScale,
                 parameters.obstacleScale,
                 parameters.forceDistance,
-                parameters.radius}};
+                parameters.radius,
+                parameters.height,
+                intoPoint(parameters.head_position),
+                intoPoint(parameters.head_velocity),
+                parameters.shoulder_rotation_angle_z,
+                parameters.shoulder_rotation_velocity_z,
+                parameters.trunk_rotation_angle_x,
+                parameters.trunk_rotation_velocity_x,
+                parameters.trunk_rotation_angle_y,
+                parameters.trunk_rotation_velocity_y,
+                intoPoint(parameters.heel_right_position),
+                intoPoint(parameters.heel_right_velocity),
+                intoPoint(parameters.heel_left_position),
+                intoPoint(parameters.heel_left_velocity)
+            }};
         result = simulation->AddAgent(std::move(agent));
     } catch(const std::exception& ex) {
         if(errorMessage) {
```

**Reply by ThoChat** [reply]
Awsome!!
Thank you so much! This was driving me crazy.
I was too focused on finding something in the libsimulator folder.

**Comment by ThoChat** [comment]
@schroedtert and @ChristianHirt maybe you could have an idea too ?
This is coming from the initiation of the parameter in the simulation core, but we couldn't figure out where...
I'm kinda blocked now because of this.
